### PR TITLE
RC_Channel: Aux switches to respect 'reverse' option

### PR DESCRIPTION
**Pull request by SergeyBokhantsev**

### CODE REVIEW

**Comment by tridge** [comment]
we probably should have done this when we first introduced the feature. Unfortunately if we apply this now then it could be dangerous. eg. someone may have a RC option setup for arming and the vehicle may arm when they upgrade firmware.
@peterbarker, what do you think?

**Comment by IamPete1** [comment]
I think this is useful to have, we could go through and set all switch reversing params to zero as part of this.

**Comment by rmackay9** [comment]
We also don't use the RCx_MIN/MAX or _TRIM values and a similar issue exists for the flight mode switch.

My feeling at the moment is that this change, while certainly valid if we'd done it originally, will cause more problem than it will resolve if we introduce it now.

**Comment by peterbarker** [comment]
As @tridge points out, there will be transition issues.

@IamPete1 suggests using `configured_in_storage` - but someone may have explicitly set those parameters themselves - and we currently ignore them but would stop, leading to potential arming attempts.

@rmackay9 points out we don't use the min/max/trim parameters; it's unclear to me how we would do that since AFAIK aux and mode switches have always been done on raw PWM threshold values.  Would be a rather interesting documentation problem, if nothing else, if you were to attempt to apply the parameters to the mode switch!

What problem are we attempting to solve here?  I'm guessing some sort of non-programmable transmitter?!

**Comment by SergeyBokhantsev** [comment]
@peterbarker the problem is the same as one solved by having 'reverse' option for channel value, I believe. It looks like there is no way to safe handle this change, so it is easier leave it as is.
I'm not forcing this in any way.

**Comment by tridge** [comment]
we could add a bit in RC_OPTIONS which says "honor the reversed flag on switches"

**Comment by SergeyBokhantsev** [comment]
@tridge done with RC_OPTIONS

**Comment by IamPete1** [comment]
@SergeyBokhantsev Thanks for updating this, sorry it has taken us so long to get to it. It would be great if you could rebase it.

**Comment by tridge** [comment]
I have rebased this

**Comment by peterbarker** [comment]
Ah.  I just had a thought.

Some options are considered "high" when they're in the middle.  We should consider how that's handled when the switch is reversed.

**Comment by peterbarker** [comment]
avoidance and OSD control are two examples where behaviour isn't strictly reversed by this flag.

**Comment by andyp1per** [comment]
I think for OSD control it's only actual switches where this is true - the stick commands use high/low. So I think it's fine.

**Comment by SergeyBokhantsev** [comment]
Mid-position behavior does not change with or without the reverse. I believe this is ok. How it may be a problem?

File: libraries/RC_Channel/RC_Channel.h [hunk 1]
```diff
@@ -358,6 +358,11 @@ class RC_Channels {
         return get_singleton() != nullptr && (_options & uint32_t(Option::FPORT_PAD));
     }
 
+    // should a channel reverse option affect aux switches
+    bool switch_reverse_allowed(void) const {
+        return get_singleton() != nullptr && (_options & uint32_t(Option::ALLOW_SWITCH_REV));
```
Comment (tridge) [inline libraries/RC_Channel/RC_Channel.h:363]: i think we can remove the get_singleton() check on all these functions, it looks like it was leftover from when we needed singleton to get at _options

**Pull request merged**